.circleci/config.yml

```diff
@@ -150,17 +146,19 @@ jobs:
             && cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_COVERAGE=On -DCPACK_GENERATOR=DEB \
                 -DENABLE_COMPILER_WARNINGS=On -DENABLE_WERROR=Off -DCMAKE_EXPORT_COMPILE_COMMANDS=On
       - run: make -C build -j4
-      - run: scripts/clang-tidy-only-diff.sh 4
       - run: make -C build -j2 tests
       - run: make -C build -j2 check
+      # Note: we save the cache here before doing linting so that if linting fails, we can rebuild quickly
```

Comment:
Should we have a separate build target for linting? Many times linting is the culprit and having to wait for a build, even if partially cached, would waste some time.

Reply:
Yeah, it's the slowest part - I need to do more thinking about how we can refactor the CI workflow to not get blocked by that.